src/migrations/20220819014844-user-settings-table.js

```diff
@@ -0,0 +1,76 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => queryInterface.sequelize.transaction(async (transaction) => {
+    await queryInterface.createTable(
+      'UserSettings',
+      {
+        id: {
+          type: Sequelize.INTEGER,
+          primaryKey: true,
+          autoIncrement: true,
+          allowNull: false,
+        },
+        // 'email', etc.
+        class: { type: Sequelize.STRING, allowNull: false },
+        key: { type: Sequelize.STRING, allowNull: false },
+        default: { type: Sequelize.JSONB, allowNull: false },
+        createdAt: { allowNull: false, type: Sequelize.DATE },
+        updatedAt: { allowNull: false, type: Sequelize.DATE },
+      },
+      { transaction },
+    );
+
+    const keys = [
+      'emailWhenReportSubmittedForReview',
+      'emailWhenChangeRequested',
+      'emailWhenReportApproval',
+      'emailWhenAppointedCollaborator',
+    ];
+
+    await Promise.all(keys.map(async (key) => {
+      await queryInterface.sequelize.query(
+        `
+        INSERT INTO "UserSettings" ("class", "key", "default", "createdAt", "updatedAt")
+        VALUES ('email', '${key}', '"never"', current_timestamp, current_timestamp)
+        `,
+        { transaction },
+      );
+    }));
+
+    await queryInterface.createTable(
+      'UserSettingOverrides',
+      {
+        id: {
+          type: Sequelize.INTEGER,
+          allowNull: false,
+          primaryKey: true,
+          autoIncrement: true,
+        },
+        userId: {
+          allowNull: false,
+          type: Sequelize.INTEGER,
+          references: { model: { tableName: 'Users' }, key: 'id' },
+        },
+        userSettingId: {
+          allowNull: false,
+          type: Sequelize.INTEGER,
+          references: { model: { tableName: 'UserSettings' }, key: 'id' },
+        },
+        value: { allowNull: false, type: Sequelize.JSONB },
+        createdAt: { allowNull: false, type: Sequelize.DATE },
+        updatedAt: { allowNull: false, type: Sequelize.DATE },
+      },
+      { transaction },
+    );
+
+    await queryInterface.addIndex('UserSettingOverrides', ['userId', 'userSettingId'], { transaction });
+    await queryInterface.addConstraint('UserSettingOverrides', {
+      fields: ['userId', 'userSettingId'],
+      type: 'unique',
+      transaction,
+    });
+  }),
+  down: (queryInterface) => queryInterface.sequelize.transaction(async (transaction) => {
+    await queryInterface.dropTable('UserSettingOverrides', { transaction });
+    await queryInterface.dropTable('UserSettings', { transaction });
+  }),
```

Comment:
In this case, we do want the "down" to be functional, so we need to add the removal of the ZALUserSetting and ZALUserSettingOverrides and functions for production.

Previously, we made an exception to omit the "down" since it wasn't very feasible to write it, but it was in a v. lengthy and complex migration.

Reply:
Got it! Restored this here: 7d4fe57